paramiko/config.py

```diff
@@ -233,7 +233,7 @@ def lookup(self, hostname):
         if canon and hostname.count(".") <= maxdots:
             # NOTE: OpenSSH manpage does not explicitly state this, but its
             # implementation for CanonicalDomains is 'split on any whitespace'.
-            domains = options["canonicaldomains"].split()
+            domains = options["canonicaldomains", ""].split()
```

Comment:
If you're going for a fallback to the empty string, it should be:

```suggestion
            domains = options.get("canonicaldomains", "").split()
```